providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -33,6 +37,7 @@ public class FlagdProvider extends EventProvider implements FeatureProvider {
 
     private EvaluationContext evaluationContext;
 
+    private Map<String, FeatureFlag> changedFlags = new HashMap<>();
```

Comment:
Keeping this state is tricky, and will almost certainly lead to bugs. It's also not threadsafe.

I think instead of keeping this state in a few places, you shoud add an additional nullable parameter to the stateSupplier functions which has the changed flags - then you can just use that in the event emission. That way there's no state to manage or synchronize or worry about.

Reply:
understood, will do this

Reply:
Have fixed it, can you please help with review 😅